File: retinaguard-backend/scripts/verify_preprocessing.py
import sys
import os
import json
import torch
import cv2

sys.path.insert(0, r"D:\Retina-Guard\model\RetinaGuard_ML")
from src.preprocessing import retina_guard_baseline_preprocess

def run():
    file = sys.argv[1]
    
    # Original dims
    img = cv2.imread(file)
    if img is None:
        print("Failed to read image")
        sys.exit(1)
        
    h, w, c = img.shape
    
    tensor = retina_guard_baseline_preprocess(file)
    
    arr = tensor.numpy()
    
    print(json.dumps({
        "engine": "python",
        "originalDimensions": [w, h],
        "processedDimensions": list(arr.shape[1:]),
        "dtype": str(arr.dtype),
        "channelOrder": "RGB",
        "tensorShape": list(arr.shape),
        "min": float(arr.min()),
        "max": float(arr.max()),
        "mean": float(arr.mean()),
        "std": float(arr.std())
    }, indent=2))

if __name__ == "__main__":
    run()

File: retinaguard-backend/scripts/verify_preprocessing.js
const { preprocessImage } = require('../src/inference/preprocess');

async function run() {
    const file = process.argv[2];
    const { tensor: arr, metadata: info } = await preprocessImage(file);
    
    let min = arr[0], max = arr[0], sum = 0;
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] < min) min = arr[i];
        if (arr[i] > max) max = arr[i];
        sum += arr[i];
    }
    const mean = sum / arr.length;
    let sumSq = 0;
    for (let i = 0; i < arr.length; i++) {
        sumSq += Math.pow(arr[i] - mean, 2);
    }
    const std = Math.sqrt(sumSq / arr.length);

    console.log(JSON.stringify({
        engine: 'node',
        originalDimensions: [info.inputWidth, info.inputHeight],
        processedDimensions: info.shape.slice(2),
        dtype: info.dtype,
        channelOrder: 'RGB',
        tensorShape: info.shape,
        min, max, mean, std
    }, null, 2));
}

run().catch(console.error);
